### PR TITLE
refactor: course filters

### PR DESCRIPTION
**Pull request by MaxFrank13**
This refactors the course filters functionality to instead use redux to manage the state for the filters. Previously, this was done through the use of the `useCheckboxSetValues` hook from Paragon. This hook is used to easily manage the state of a checkbox set. By having the state management through redux, we are enabled to modify these filters another way (i.e. without using the checkboxes).

To be clear, this work does not change anything about the behavior of the Learner Dashboard and it's filter component. This only changes the technical implementation for how that behavior is managed.

Also added in here is some documentation for some of the functions that were modified.

### CODE REVIEW

File: src/data/redux/app/reducer.js [hunk 1]
```diff
@@ -10,15 +10,17 @@ const initialState = {
   enterpriseDashboard: {},
   platformSettings: {},
   suggestedCourses: [],
-  filterState: {},
```
Comment (jsnwesson) [inline src/data/redux/app/reducer.js:13]: huh, was this not being used anywhere?

File: src/data/redux/hooks/app.js [hunk 2]
```diff
@@ -7,8 +7,11 @@ import * as module from './app';
 const selectors = redux.selectors.app;
 const actions = redux.actions.app;
 
+// TODO: docs: this entire file
```
Comment (jsnwesson) [inline src/data/redux/hooks/app.js:10]: did you intend to keep this here for a later ticket?

**Review by jsnwesson — APPROVED** [review]
looks good to me, thanks for cleaning this up with documentation and removing a layer of abstraction.

File: src/data/redux/app/reducer.js (hunk 1 above)
Comment (MaxFrank13) [reply src/data/redux/app/reducer.js:13]: I could not find any reference to it

File: src/data/redux/hooks/app.js (hunk 2 above)
Comment (MaxFrank13) [reply src/data/redux/hooks/app.js:10]: Good catch! I'll remove this

**Pull request merged**